devicetypes/smartthings/zigbee-thermostat.src/zigbee-thermostat.groovy

```diff
@@ -109,92 +116,181 @@ metadata {
 		valueTile("battery", "device.battery", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
 			state "battery", label:'${currentValue}% battery', unit:""
 		}
-		main "temperature"
-		details(["temperature", "lowerHeatingSetpoint", "heatingSetpoint", "raiseHeatingSetpoint", "lowerCoolSetpoint",
-				 "coolingSetpoint", "raiseCoolSetpoint", "thermostatMode", "thermostatFanMode", "thermostatOperatingState",
-				 "refresh", "battery", "powerSource"])
+		main "thermostatMulti"
+		details(["thermostatMulti", "thermostatMode", "heatingSetpoint", "coolingSetpoint", "thermostatFanMode", "battery", "powerSource", "refresh"])
 	}
 }
 
 def parse(String description) {
 	def map = zigbee.getEvent(description)
 	def result
-	if(!map) {
+
+	if (!map) {
 		result = parseAttrMessage(description)
 	} else {
 		log.warn "Unexpected event: ${map}"
 	}
+
 	log.debug "Description ${description} parsed to ${result}"
+
 	return result
 }
 
 private parseAttrMessage(description) {
```

Comment:
This function is a little long. It might make sense to break out the individual attribute handling into separate functions.